source/session_encrypt.c

```diff
@@ -168,12 +168,10 @@ static int build_header(struct aws_cryptosdk_session *session, struct aws_crypto
 }
 
 static int sign_header(struct aws_cryptosdk_session *session) {
-    AWS_PRECONDITION(session != NULL);
-    AWS_PRECONDITION(aws_allocator_is_valid(session->alloc));
-    AWS_PRECONDITION(aws_cryptosdk_hdr_is_valid(&session->header));
+    AWS_PRECONDITION(aws_cryptosdk_session_is_valid(session));
+    AWS_PRECONDITION(session->alg_props->impl->cipher_ctor != NULL);
     AWS_PRECONDITION(session->header.iv.len <= session->alg_props->iv_len);
     AWS_PRECONDITION(session->header.auth_tag.len <= session->alg_props->tag_len);
-    AWS_PRECONDITION(aws_cryptosdk_sig_ctx_is_valid(session->signctx));
```

Comment:
Not a blocker, but I'm curious why this is no longer needed?

Reply:
The new session validator checks that `session->signctx` is either valid or `NULL`. The function does actually check before using it on `aws_cryptosdk_sig_update` so it was probably not required before and by using the session validator we are considering more cases.

Reply:
@alex-chew we have investigated this issue and we have decided to remove the assumption from the harness. Note that the session validator allows `session->signctx` to be either NULL or valid. To do so, we modified a precondition in `EVP_DigestUpdate` which we think is too restrictive.

Moreover, we have opened an issue (#685) to remove this type of validators from the code base, since they cannot be mirrored as preconditions.